core/src/main/java/org/mule/runtime/core/api/util/StreamingUtils.java

```diff
@@ -104,33 +105,73 @@ public CoreEvent handleException(Throwable throwable) {
+   * {@link CursorProviderFactory#of(EventContext, Object, ComponentLocation)} is returned. Otherwise, the original {@code value} is.
+   *
+   * @param value                 a value which may be a repeatable streaming resource
+   * @param cursorProviderFactory a nullable {@link CursorStreamProviderFactory}
+   * @param eventContext          the root context of the event on which the {@code value} was generated
+   * @return the {@code value} or a {@link CursorProvider}
+   * 
+   * @deprecated Use {@link #streamingContent(Object, CursorProviderFactory, EventContext, ComponentLocation)}
+   */
+  @Deprecated
+  public static Object streamingContent(Object value, CursorProviderFactory cursorProviderFactory, EventContext eventContext) {
+    return streamingContent(value, cursorProviderFactory, eventContext, eventContext.getOriginatingLocation());
+  }
+
+  /**
+   * If the {@code cursorProviderFactory} accepts the given {@code value}, then the result of invoking
+   * {@link CursorProviderFactory#of(EventContext, Object, ComponentLocation)} is returned. Otherwise, the original {@code value} is.
    *
    * @param value                 a value which may be a repeatable streaming resource
    * @param cursorProviderFactory a nullable {@link CursorStreamProviderFactory}
    * @param event                 the event on which the {@code value} was generated
+   * @param originatingLocation   the {@link ComponentLocation} where the cursor was created
    * @return the {@code value} or a {@link CursorProvider}
+   *
+   * @since 4.4.0
    */
+  public static Object streamingContent(Object value, CursorProviderFactory cursorProviderFactory, CoreEvent event,
+                                        ComponentLocation originatingLocation) {
+    return streamingContent(value, cursorProviderFactory, getRoot(event.getContext()),
+                            originatingLocation);
+  }
+
+  /**
+   * If the {@code cursorProviderFactory} accepts the given {@code value}, then the result of invoking
+   * {@link CursorProviderFactory#of(EventContext, Object, ComponentLocation)} is returned. Otherwise, the original {@code value} is.
+   *
+   * @param value                 a value which may be a repeatable streaming resource
+   * @param cursorProviderFactory a nullable {@link CursorStreamProviderFactory}
+   * @param event                 the event on which the {@code value} was generated
+   * @return the {@code value} or a {@link CursorProvider}
+   *
+   * @deprecated Use {@link #streamingContent(Object, CursorProviderFactory, EventContext, ComponentLocation)}
+   */
+  @Deprecated
   public static Object streamingContent(Object value, CursorProviderFactory cursorProviderFactory, CoreEvent event) {
-    return streamingContent(value, cursorProviderFactory, getRoot(event.getContext()));
+    return streamingContent(value, cursorProviderFactory, event, event.getContext().getOriginatingLocation());
```

Comment:
can you make the implementation null safe and pass no value instead of a wrong value for this case?